### PR TITLE
service: fix failure on transition to readying

### PR DESCRIPTION
**Pull request by dirkmueller**
When the salt barclamp is used, a transition to readying is failing
with this error:

  salt transition to default failed.
  wrong number of arguments (3 for 0)
  /opt/dell/crowbar_framework/app/models/service_object.rb:295:in `transition'
  /opt/dell/crowbar_framework/app/models/crowbar_service.rb:202:in `block (2 levels) in transition'
  /opt/dell/crowbar_framework/app/models/crowbar_service.rb:186:in `each'
  /opt/dell/crowbar_framework/app/models/crowbar_service.rb:186:in `block in transition'
  /opt/dell/crowbar_framework/app/models/crowbar_service.rb:185:in `each'
  /opt/dell/crowbar_framework/app/models/crowbar_service.rb:185:in `transition'
  /opt/dell/crowbar_framework/app/controllers/barclamp_controller.rb:126:in `transition'

This is due to incorrect number of arguments being specified on the
base service_object

Please help potential reviewers to understand this pull request and speed
up the process by writing a meaningful pull request message.

Answering the following questions can help, but is optional.

**Why is this change necessary?**

**How does it address the issue?**

**Is there additional information worth sharing like links to a Trello
card, bug references, testing advice or dependencies to other pull
requests?**

### CODE REVIEW

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -292,7 +292,8 @@ def versions
     [200, { versions: ["1.0"] }]
   end
 
-  def transition
+  def transition(inst, name, state)
+    Rails.logger.debug("transition: enter #{inst} #{name} #{state}")
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: Please don't add just a "enter" log line, as we'll be looking for the "leave" log line. Can you either drop the verb, or make it "enter and leave"?

**Comment by vuntz** [comment]
Btw, while this is indeed a bug, this is also a bug in the salt barclamp because I pretty much doubt the transition method should be called for it. That means it has some wrong metadata...

**Comment by dirkmueller** [comment]
@nicolasbock the salt barclamp seems to be the first one not reimplementing this function

**Comment by nicolasbock** [comment]
@dirkmueller That's sneaky :smile:

**Pull request merged**